### ISSUE TITLE
Issue with peer dependency check on v1.5.2.2

### ISSUE THREAD

**Issue by theosp** [meteor/meteor#9235]
Following upgrade to Meteor v1.5.2.2, https://github.com/tmeasday/check-npm-versions starts to report that the **mysql** package can't be found even though ```console.log(require("mysql"));``` works fine.

The issue is due to [check-npm-versions](https://github.com/tmeasday/check-npm-versions/blob/master/check-npm-versions.js#L10) failure to load ```require("mysql/package.json")``` .

Running the following in ```$ meteor shell``` works fine on v1.5.2 (the version I upgraded from) but fails on v1.5.2.2 .

```require("mysql/package.json")```.

Other packages that I use check-npm-versions to verify their existence works well (e.g. bluebird), so for me the issue is specific to mysql.

Any clue for what cause this regression?

**Comment by benjamn** [comment]
Where is the `mysql` npm package installed in your environment?

**Comment by theosp** [comment]
Under the node_modules package in the app directory.

package.json:
```
{
  "name": "justdo-landing-app",
  "private": true,
  "scripts": {
    "start": "meteor run"
  },
  "dependencies": {
   ...
    "mysql": "~2.13.0",
   ...
  }
}
```

**Comment by theosp** [comment]
Let me know if you want me to come up with a demo app for this issue.

**Comment by benjamn** [comment]
Ok, so, the npm `mysql` package is somewhat unusual because it does not have a `"main"` field in its `package.json` file, and instead just has an `index.js` file. This means Meteor's `ImportScanner` and `Resolver` do not need any help from the `mysql/package.json` file to import `mysql`, so it doesn't have to be included in the JS bundle.

The `check-npm-versions` package attempts to import the `mysql/package.json` module [dynamically](https://github.com/tmeasday/check-npm-versions/blob/f50c9860d820b856c353e93590a3cf33339ea2b8/check-npm-versions.js#L10), which works just fine if the `package.json` file (or even a small subset of it) is in the JS bundle, but fails if the `package.json` file was omitted from the bundle. In other client-side module systems, dynamic imports typically don't work at all, but in Meteor they work as long as the module happens to be included in the bundle.

In short, if you want to use `check-npm-versions` for `mysql`, you should make sure `require("mysql/package.json")` appears somewhere in your code (it doesn't even have to be executed), so that the bundler is aware of the dependency.

**Comment by hwillson** [comment]
On a related note, I'll just drop these here since I was looking this at the exact same time @benjamn posted 🙂:

- Ben's great detailed explanation of this: https://github.com/meteor/meteor/pull/8327#issuecomment-280876264 (you might have to find the "load more" on that page and load it, before you can see the comment, due to the size of that thread)
- Open feature request tracking dynamic imports: https://github.com/meteor/meteor-feature-requests/issues/167

**Comment by benjamn** [comment]
Perhaps we should include a stub for `package.json` when it exists, even if it isn't used for package resolution. With the workaround suggested above, the whole `package.json` file gets bundled, which is not necessary to check the package version.
